### DISCUSSION TITLE
en/latest/UserGuide/%E6%95%B0%E6%8D%AE%E8%A6%81%E6%B1%82/

### DISCUSSION THREAD

**Discussion by giscus** [zdsjjtTLG/TrackIt#10] [General]
# en/latest/UserGuide/%E6%95%B0%E6%8D%AE%E8%A6%81%E6%B1%82/

在python中轻松进行路径匹配

https://gotrackit.readthedocs.io/en/latest/UserGuide/%E6%95%B0%E6%8D%AE%E8%A6%81%E6%B1%82/

**Comment by Zeitreisender11** [comment]
请问一下，gotrackit是只支持dir=0双向，和dir=1单向的线层么，可以支持处理反向的线层么，即我的原始路网线层有3个方向属性：双向，正向，反向，这种情况怎么处理比较好呢？

**Comment by zdsjjtTLG** [comment]
目前不支持，我个人比较反感这种dir=-1的，觉得非常不方便，其实反转一下就行了：

![image](https://github.com/user-attachments/assets/1b72e21d-4958-4912-abf9-c3c8534533ff)

**Reply by zdsjjtTLG** [reply]
看来还是有需求，下一版本会支持dir=-1的情况

**Reply by zdsjjtTLG** [reply]
反转后将dir直接改为1就好了

**Reply by Zeitreisender11** [reply]
请问一下，我想将路的方向属性dir加入匹配到的match_link.geojson中的"properties"中，该怎么做？
顺便问一下，下个支持dir=-1的版本什么时候更新？
感谢！

**Reply by zdsjjtTLG** [reply]
用geopandas读取进来当作dataframe一样操作就行，下一个版本就会更新dir=-1的情况

**Reply by wuhao-ai** [reply]
怎么在路网文件中加入geometry字段？

**Reply by zdsjjtTLG** [reply]
shp或geojson等地理信息文件都是自带geometry字段的，你要可视化编辑这个geometry可以使用QGIS，代码操作geometry可以使用geopandas和shapely

**Comment by ChristopherChou** [comment]
请问一下，我从omsnx下载的数据与要求的路网数据字段不一致，应该如何处理

**Reply by ZheChengData** [reply]
https://github.com/zdsjjtTLG/TrackIt/discussions/5

**Reply by ZheChengData** [reply]
这个问题有人问过了，你看看上面这个链接

**Reply by ChristopherChou** [reply]
感谢解答，FMM是支持直接对轨迹序列进行匹配的，后续会考虑增加轨迹匹配的功能吗

**Reply by zdsjjtTLG** [reply]
现在的输入就是轨迹序列呀，GPS点序列不就是轨迹序列嘛。还是说，你的意思是轨迹线？

**Reply by zdsjjtTLG** [reply]
FMM**也不是对轨迹线进行直接匹配的**，也是把线拆成点来匹配的，这个很简单，如果你的轨迹是LinString形式的话，用下面这个函数转化为点(然后自己赋予一个时间列就好了)，再用gotrackit进行匹配：

```
geometry_field = 'geometry'
lng_field = 'lng'
lat_field = 'lat'

def line2gps(path_gdf: gpd.GeoDataFrame = None):
    path_gdf[geometry_field] = path_gdf[geometry_field].apply(lambda line: list(line.coords))
    gps_df = pd.DataFrame(path_gdf).explode(column=[geometry_field], ignore_index=True)
    gps_df[lng_field] = gps_df[geometry_field].apply(lambda t: t[0])
    gps_df[lat_field] = gps_df[geometry_field].apply(lambda t: t[1])
    del gps_df[geometry_field]
    return gps_df
```
